client/packages/common/src/ui/layout/tables/components/Cells/DotCell/DotCell.tsx

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+import { CellProps } from '../../../columns';
+import { CircleIcon } from '@openmsupply-client/common';
+import { RecordWithId } from '@common/types';
+
+export const DotCell = <T extends RecordWithId>({
+  rowData,
+  column,
+}: CellProps<T>): React.ReactElement<CellProps<T>> => {
+  return (
+    <>
+      {column.accessor({ rowData }) === true && (
+        <CircleIcon sx={{ color: 'black', transform: 'scale(0.5)' }} />
+      )}
+    </>
+  );
+};
```

Comment:
:nit

all good as it is.. just a couple of observations. 
I avoid using hard-coded colour names, and have tried to avoid using hard coded hex or RGB values too, in favour of theme colours, so that colours can be updated with a custom theme. In this case - I like the not-quite-black that we have for typography, it's a slightly softer feel having a dark grey than an absolute black.

The use of `transform` is very nice! It made me wonder what would happen if the SVG was a different size.. but that can't actually happen, as the mui svg is fixed at 24px. I would usually add a hard-coded width and/or height; I think the transform is neater - nice to have it flexible.

I use the shortcut of not using `{ return` in anonymous methods - I have no argument for it other than it uses slightly fewer characters and is neater 🤷 

I would also use a ternary operator here (? :) to explicitly say that we're returning a circle icon or `null` i.e. nothing is rendered. I think that's clearer what the intention is. I would just use `&& <Element..` when the whole block is JSX

```suggestion
export const DotCell = <T extends RecordWithId>({
  rowData,
  column,
}: CellProps<T>) =>
  column.accessor({ rowData }) === true ? (
    <CircleIcon
      sx={{
        color: theme => theme.typography.body1.color,
        transform: 'scale(0.5)',
      }}
    />
  ) : null;
```

Reply:
That is awesome, thank you for the detailed feedback, Mark :)

Reply:
I wonder if you would be willing to explain this line?
It feels like it should be very basic knowledge, but I don't really get how it is calling theme without importing it, or why you can't simply write:

```
color: theme.typography.body1.color,
```

Reply:
it's non-intuitive magic coming from the material ui [sx prop](https://mui.com/system/getting-started/the-sx-prop/)
This is a way of setting styles, but with some handy shortcuts. For example - previously when accessing things directly from the theme, you'd have to use something like the `useTheme()` hook in order to access the global theme. In the sx prop you can pass a string property (`color: '#ff00ff'`) or you can pass in a function which has a single paramter, which is the theme (`color: theme => theme.typography.body1.color`)

this one is handy if you want to use customisations to the base theme, like we've done.

The other one I use all the time : instead of specifying a hard coded colour, for CSS properties that expect a colour (color, backgroundColor..), you can pass in the path in the theme (have a look at `theme.ts` to see what's available). So, you could say `color: '#e95c30'` or you can do `color: 'primary.main'` and the sx parsing will replace 'primary.main' with the colour value from `theme.palette.primary.main`. More on [color here](https://mui.com/system/getting-started/the-sx-prop/#palette)

One other thing I use a lot - in material-ui land, there's a predefined values for spacing (default: 8px). When specifying the `style` prop, you can have `padding` as a number or a string.. if it's a number then it converts that to 'px' for you. So `paddingTop: 8` is the same as `paddingTop: '8px'`. That doesn't behave the same in `sx` though, you have to specify '8px' if you want pixels. What you can do though, is provide a number and that is multiplied by the default padding value. It means that you can just say `paddingTop: 1` and that gives you the 8px. You can then adjust spacing for the whole site by just updating `theme.spacing`. [More on spacing here](https://mui.com/system/getting-started/the-sx-prop/#spacing)

More on that in the [theme aware properties](https://mui.com/system/getting-started/the-sx-prop/#theme-aware-properties) section